### PR TITLE
Add coverage tests and http retry and error on GetTile http error

### PR DESCRIPTION
**Pull request by siiinc**
## What does this PR do?
Add missing coverage tests for template api mode
Add http max_retry=3 and retry_delay=1s
Report error on GetTile failure

## Tasklist

 - [ ] ADD YOUR TASKS HERE
 - [ ] Add test case(s)
 - [ ] Review
 - [ ] Adjust for comments
 - [ ] All CI builds and checks have passed

## Environment

Provide environment details, if relevant:

* OS:
* Compiler:

### CODE REVIEW

File: autotest/gdrivers/rda.py [hunk 1]
```diff
@@ -691,6 +691,274 @@ def rda_graph_nominal():
...
+    handler = webserver.SequentialHandler()
+    handler.add('POST', '/auth_url', 200, {}, '{"access_token": "token", "expires_in": 3600}')
+    handler.add('GET', '/rda_api/metadata/foo/bar/metadata.json', 200, {}, json.dumps(metadata_json))
+    handler.add('GET', '/rda_api/tile/foo/bar/0/0.tif', 502, {}, json.dumps({"Error": "502"}))
+
+    try:
+        org_use_exceptions = gdal.GetUseExceptions()
+        if not org_use_exceptions:
+            gdal.UseExceptions()
+        gdal.PushErrorHandler('CPLQuietErrorHandler')
```
Comment (rouault) [inline autotest/gdrivers/rda.py]: complicated way of silencing errors. You can now use
```
with gdaltest.error_handler():
       some code that might output GDAL errors / warnings
```

File: gdal/frmts/rda/rdadataset.cpp [hunk 2]
```diff
@@ -920,6 +927,10 @@ json_object* GDALRDADataset::ReadJSonFile(const char* pszFilename,
                 osURL += std::get<0>(tup)+"="+std::get<1>(tup);
                 nCountOptions ++;
             }
+            if(osURL.endsWith("&"))
```
Comment (rouault) [inline gdal/frmts/rda/rdadataset.cpp]: how could this happen now ?

File: gdal/frmts/rda/rdadataset.cpp [hunk 3]
```diff
@@ -1688,8 +1699,21 @@ void GDALRDADataset::BatchFetch(int nXOff, int nYOff, int nXSize, int nYSize)
...
         for( int j = 0; j < nToDownload; j++ )
         {
-            if( pasResults[j]->pabyData )
+            if( pasResults[j]->pszErrBuf != nullptr )
+            {
+                CPLString url = apszURLLists[i+j];
+                CPLString message = pasResults[j]->pabyData ?
+                                    CPLSPrintf("%s: %s", pasResults[j]->pszErrBuf,
+                                               reinterpret_cast<const char*>(pasResults[j]->pabyData )) :
+                pasResults[j]->pszErrBuf;
```
Comment (rouault) [inline gdal/frmts/rda/rdadataset.cpp]: should be intended at the same level as the above CPLSPrintf(), which could be just 4 characters right of the CPLString message declaration

File: autotest/gdrivers/rda.py (hunk 1 above)
Comment (siiinc) [reply autotest/gdrivers/rda.py]: Done

File: gdal/frmts/rda/rdadataset.cpp (hunk 2 above)
Comment (siiinc) [reply gdal/frmts/rda/rdadataset.cpp]: removed

File: gdal/frmts/rda/rdadataset.cpp (hunk 3 above)
Comment (siiinc) [reply gdal/frmts/rda/rdadataset.cpp]: replaced

**Pull request merged**